File: chat-resource/src/main/java/org/qbynet/chat/repository/MemberRepository.java
package org.qbynet.chat.repository;

import org.qbynet.chat.entity.Conversation;
import org.qbynet.chat.entity.Member;
import org.qbynet.chat.entity.User;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface MemberRepository extends MongoRepository<Member, String> {
    Optional<Member> findByUserAndConversation(User user, Conversation conversation);
}

File: src/main/kotlin/org/cubewhy/qbychat/repository/MemberRepository.kt
package org.cubewhy.qbychat.repository

import org.cubewhy.qbychat.entity.Member
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux

@Repository
interface MemberRepository : ReactiveMongoRepository<Member, String> {
    fun findAllByUser(user: String): Flux<Member>
    fun findAllByChat(chat: String): Flux<Member>
}